packages/react-core/src/hooks/useDataState.ts

```diff
@@ -1,42 +1,50 @@
 import React from 'react';
 
-interface ActionState<T> {
+export interface DataState<T> {
   data: T;
+  hasError: boolean;
   isLoading: boolean;
   message: string | undefined;
 }
 
-const getActionState = <T>(data: T): ActionState<T> => ({
-  data,
-  isLoading: false,
-  message: undefined,
-});
+// default state
+const INITIAL_STATE = { hasError: false, isLoading: false, message: undefined };
```

Comment:
Would it make sense to have a default message for the initial state? I'm just thinking that could be useful to distinguish between an action that didn't add a message and having not performed the action yet, but I can see how it might obfuscate customer experience.

Reply:
That's an interesting proposal! Will consider it in the future